### PR TITLE
Add BigDipper role

### PR DESCRIPTION
**Pull request by RangerMauve**
TODO (taken over by @makeworld-the-better-one)

- [x] Import postgres schemas
- [x] Add scripts for copying genesis file
- [x] Get fully working
- [x] Inventory file
- [x] Get theta working
- [x] Go through TODOs in files
- [x] Set up letsencrypt TLS reverse-proxy optionally
- [x] Check for security issues: exposed endpoints, public passwords etc
- [x] Get transactions working in Theta
- [x] Basic docs

### CODE REVIEW

**Comment by makeworld-the-better-one** [comment]
Current issue is with parsing the genesis file.

Command:

```
export PATH=$PATH:/opt/go/1.18.4/bin:/root/go/bin
bdjuno parse genesis-file --home /home/bigdipper/.bdjuno --genesis-file-path /home/bigdipper/.bdjuno/genesis.json
```

Output:

```
4:25PM DBG parsing genesis module=auth
4:28PM DBG parsing genesis module=consensus
4:28PM DBG parsing genesis module=gov
ERROR: error while storing genesis governance proposals: error while storing deposits: pq: insert or update on table "proposal_deposit" violates foreign key constraint "proposal_deposit_height_fkey"
```

**Comment by makeworld-the-better-one** [comment]
Opened some issues/PRs:
- https://github.com/forbole/bdjuno/issues/449
- https://github.com/forbole/big-dipper-2.0-cosmos/pull/959

These aren't blockers. The issue they address is ports being open to the public when they shouldn't be, but I don't think it's a security issue.

**Comment by makeworld-the-better-one** [comment]
I've left the server running as a live demo for review purposes. URL is in chat.

**Comment by dasanchez** [comment]
@makeworld-the-better-one 
What's the reason for requiring the `ansible-galaxy install -r requirements.yml` command now?

**Comment by makeworld-the-better-one** [comment]
@dasanchez the bigdipper makes use of some third-party Ansible deps/modules that need to be installed. In earlier iterations of this PR, these deps were used by the gaia role as well (for Go installation). Now that only the bigdipper roles needs them, I *think* the command could be left out of documentation unless bigdipper was being run. I've left it in because it'll be useful in the future. As I understand it, the plan is to use hypha.common to replace other roles, so deps will need to be installed anyway.

**Comment by dasanchez** [comment]
@makeworld-the-better-one it looks like the proposals are being pulled from the wrong source in the test server, these are the ones that we expect to see:
https://explorer.theta-testnet.polypore.xyz/proposals
It is probably related to the validator names display too

File: roles/bigdipper/templates/bigdipper-general_config.json.j2 [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "maintainer": {
+    "name": "Hypha Worker Co-operative",
+    "url": "https://hypha.coop/"
+  },
+  "github": {
+    "reportIssue": "https://github.com/hyphacoop/cosmos-ansible/issues"
+  },
+  "version": "base-v2.1.0"
+}
```
[the remark this answers is not in the document]
Comment (makeworld-the-better-one) [reply roles/bigdipper/templates/bigdipper-general_config.json.j2:10]: This is a file copied from the bigdipper repo, and it doesn't end with a newline. I can add one, but JSON isn't supposed to end with a newline anyway, it's different than text files.

**Comment by makeworld-the-better-one** [comment]
Ok I will open an issue for that too. All the issues/PRs I've opened are being tracked in #124. Let me know if the proposal issues are a blocker for merging.

**Comment by makeworld-the-better-one** [comment]
- [ ] Get this step working consistently (sometimes stalls): Apply metadata to hasura

File: examples/README.md [hunk 2]
```diff
@@ -138,3 +146,33 @@ Set up a Hermes relayer between two chains.
...
+    You can configure the subdomain names if needed using other variables:
+    ```yaml
+    # Defaults: 
+    hasura_host: "hasura."
+    rpc_host: "rpc."
+    bdui_host: ""
+    ```
+5. Install Big Dipper:
+    ```
+    ansible-playbook bigdipper.yml -i examples/inventory-bigdipper.yml -e 'target=SERVER_DOMAIN'
```
Comment (dasanchez) [inline examples/README.md:177]: I ran this line with the inventory file as-is and it stopped here:
```
TASK [bigdipper : copy genesis.json] *******************************************************************************************************************************************************************************
fatal: [{{ target }}]: FAILED! => {"changed": false, "msg": "Source /home/gaia/.gaia/config/genesis.json not found"}
```

File: examples/README.md [hunk 3]
```diff
@@ -138,3 +146,33 @@ Set up a Hermes relayer between two chains.
...
+    ansible-playbook node.yml -i examples/inventory-theta.yml -e 'target=SERVER_DOMAIN'
+    ```
+3. Go and edit [`inventory-bigdipper.yml`](inventory-bigdipper.yml) to your liking. Some things like `hasura_admin_secret` and `letsencrypt_email` will need to be changed, while others are optional.
+4. Setup DNS:
+    ```
+    mydomain.com.        3600 IN A     123.123.123.123
+    hasura.mydomain.com. 3600 IN CNAME mydomain.com.
+    rpc.mydomain.com.    3600 IN CNAME mydomain.com.
+    ```
+    You can configure the subdomain names if needed using other variables:
```
Comment (dasanchez) [inline examples/README.md:168]: Where am I setting these variables? A config file somewhere?

**Review by dasanchez — CHANGES_REQUESTED** [review]
@makeworld-the-better-one 
Can you update the playbook/inventory file and the examples readme to explain how to successfully set up a block explorer? The inventory file has several gaps that the readme section does not communicate how to fill.

File: examples/README.md [hunk 4]
```diff
@@ -138,3 +146,33 @@ Set up a Hermes relayer between two chains.
 * **Inventory file:** [`inventory-hermes.yml`](inventory-hermes.yml)
 
 Follow the [Hermes IBC Relayer Setup](/docs/Hermes-Relayer-Setup.md) guide in the `docs` folder for all the requirements and steps needed to deploy the relayer. 
+
+## Run a Big Dipper block explorer
+
+Run a block explorer for your chain using the Big Dipper [software](https://bigdipper.live/) ([docs](https://docs.bigdipper.live/)) software.
```
[the remark this answers is not in the document]
Comment (makeworld-the-better-one) [reply examples/README.md:152]: The software stack should support the Theta testnet, as well as other chains. It still has bugs as you know, but yes it is supposed to work.

File: examples/README.md (hunk 2 above)
Comment (makeworld-the-better-one) [reply examples/README.md:187]: I've specified that for this example, the blockchain node and Big Dipper installation need to be on the same machine.

**Comment by makeworld-the-better-one** [comment]
@dasanchez tried to address your review

**Comment by dasanchez** [comment]
The play stopped here:
```
TASK [bigdipper : Run docker image for Hasura] *********************************************************************************************************************************************************************************************************************
fatal: [{{ target }}]: FAILED! => {"changed": false, "msg": "Error pulling image hasura/graphql-engine:v2.9.0 - 500 Server Error for http+docker://localhost/v1.41/images/create?tag=v2.9.0&fromImage=hasura%2Fgraphql-engine: Internal Server Error (\"Get \"https://registry-1.docker.io/v2/hasura/graphql-engine/manifests/sha256:63d7e9ee630fe537338b1eda6e90407e12dc5dced9784c64fcf7e197cea9b379\": EOF\")"}
```

**Comment by makeworld-the-better-one** [comment]
Seems like fluke on Docker Hub's end. I'm able to pull the image fine just with a `docker pull` command. Let me know if get the error again.

**Comment by dasanchez** [comment]
Indeed, I ran the play once more and it didn't fail this time.

**Pull request merged**